test/testTrackPropagationPipeLine.cxx

```diff
@@ -0,0 +1,128 @@
+/// -------------------------------------------------------------------------------------+
+/// VectorFlow unit test. Generation + Propagation.                                      |
+///                                                                                      |
+/// This test is intended to show the usage of the PipelineFlow interface of VectorFlow. |
+/// The example propagates charged particles in a constant field in several steps,       |
+/// until they reach a sphere with a given radius centered in the event vertex.          |
+///                                                                                      |
+/// The pipeline flow has only one task (TaskPropagator) and the data type to be used    |
+/// are the events that are generated in the event-loop.                                 |
+/// After each execution, the pipeline is cleared and the next event is processed.       |
+/// -------------------------------------------------------------------------------------+
+
+#include <iostream>
+#include <string>
+#include <vector>
+#include <vectorFlow/PipelineFlow.h>
+#include "CocktailGenerator.h"
+#include "Event.h"
+#include "Track.h"
+#include "Particle.h"
+#include "vectorFlow/SystemOfUnits.h"
+#include "base/Vector3D.h"
+#include "HelixPropagator.h"
+#include "SimpleStepper.h"
+
+using namespace vectorflow;
+using Event_t = CocktailGenerator::Event_t;
+
+struct TaskPropagator : public Work<Event_t, std::vector<Event_t*>> {
+  // Propagator task needs a stepper
+  trackml::SimpleStepper* stepper;
+
+  // Propagate function
+  void PropagateTracks(Event_t* event) {
```

Comment:
Sorry for misleading you in the mail. The state in this case is the track, not the event. The loop on events should stay in the main(). For every event you should loop (also in main) on tracks and add them one by one in the pipeline. After that, you execute the pipeline (once per event)

Reply:
Fixed. Changed the logic so the tracks are the states and not the events. The two loops are now in main and tracks are the data input for the pipeline.